security/encryption.go

```diff
@@ -60,40 +81,60 @@ func (cd *aesGcmCrypting) validate() error {
 	return errors.New(errorsMsg.MsgErrUnknown + "." + errorsMsg.EncryptDecrypt)
 }
 
-func (cd *aesGcmCrypting) Encrypt(value []byte, additional []byte) ([]byte, error) {
-	var block, err = aes.NewCipher(cd.key)
+func (km *keyMaterial) Encrypt(value []byte, additional []byte) ([]byte, string, error) {
+	// select the most recent key
+	key := km.Keys[0]
+	var block, err = aes.NewCipher(key.Key)
 	if err != nil {
-		return nil, err
+		return nil, "", err
 	}
 
 	var iv = make([]byte, 12)
 	_, _ = rand.Read(iv)
 
 	var aesgcm cipher.AEAD
-	aesgcm, err = cipher.NewGCMWithTagSize(block, cd.tagSize)
+	aesgcm, err = cipher.NewGCMWithTagSize(block, key.tagSize)
 	if err != nil {
-		return nil, err
+		return nil, "", err
 	}
 
 	var enc = aesgcm.Seal(nil, iv, value, additional)
+	kid := key.Kid
+	encValue := append(iv, enc...)
 
-	return append(iv, enc...), err
+	return encValue, kid, err
 }
 
-func (cd *aesGcmCrypting) Decrypt(value []byte, additional []byte) ([]byte, error) {
-	if len(value) <= 12 {
+func (km *keyMaterial) Decrypt(encData []byte, kid string, additional []byte) ([]byte, error) {
+	// select the appropriate key
+	var key aesGcmKey
+	var found bool
+	for _, k := range km.Keys {
```

Comment:
A map would avoid this for loop, but we can expect really a few element so maybe wwe don't really care of it

Reply:
There would be at most 2 elements, and if we rotate the key correctly, then we will select the appropriate key as the first element. So no loop, in reality.